server/bin/build-libjxl.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -e
+
+: "${LIBJXL_REVISION:=$(jq -cr '.sources[] | select(.name == "libjxl").revision' build-lock.json)}"
+
+git clone https://github.com/libjxl/libjxl.git --recursive --shallow-submodules
+cd libjxl
+git reset --hard $LIBJXL_REVISION
+
+mkdir build
+cd build
+cmake \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DBUILD_TESTING=OFF \
+  -DJPEGXL_FORCE_SYSTEM_BROTLI=ON \
+  -DJPEGXL_FORCE_SYSTEM_HWY=ON \
+  -DJPEGXL_ENABLE_JPEGLI=OFF \
+  -DJPEGXL_ENABLE_JPEGLI_LIBJPEG=OFF \
```

Comment:
It'd be cool if we could drop libjpeg-turbo in favor of jpegli since the latter has much better compression efficiency at comparable speed. This can be done in a separate PR if you'd rather not work on it, though.

Reply:
Yeah I can try and add jpegli instead. As far as I can tell, libjxl can build jpegli and install it as libjpeg-turbo system-wide. This will definitely need extensive testing then.

Reply:
I wasn't able to get libraw working with just the JXL provided libjpeg (jerror.h was missing), so I just enabled jpegli for libjxl. The rest still uses Debian libjpeg-turbo. If I find some time I can try and have a go again at using jpegli everywhere.